src/classes/GS_NonprofitTrialOrgService.cls

```diff
@@ -0,0 +1,182 @@
+                return testingConfiguration;
+            }
+        }
+        set {
+            testingConfiguration = value;
+        }
+    }
+
+    /**
+     * @description Read-only property with the Org queried with the fields: IsSandbox and TrialExpirationDate.
+     * @return Organization object with fields IsSandbox and TrialExpirationDate.
+     */ 
+    private static Organization org { 
+        get {
+            org = getOrgInformation();
+            return org;
+        }
+        private set;
+    }
+
+     /**
+     * @description Private Constructor. This class is not expected to be instantiated.
+     */
+    private GS_NonprofitTrialOrgService() {
+    }
+
+    /**
+     * @description Return if the org is a Sandbox(Sandbox or Scratch).
+     * @return return true if org is a sandbox or scratch or the value in the field "Sandbox" of the Static Resource GettingStartedTestingConfigurationJSON.
+     */
+    public static Boolean isSandbox() {
+        return overrideValuesForTesting ? testingConfiguration.isSandbox : org.IsSandbox;
+    }
+
+    /**
+     * @description Retrieves the application date.
+     * On prod this is populated using the FeatureParameter "P10_Application_Date"
+     * For QA purpose this values can be overriden if this is running on a Scratch org and there is a value in the Static Resource GettingStartedTestingConfigurationJSON
+     * @return   return Application date gotten of the P10_Application_Date FeatureParameter OR the testing configuration object.
+     */
+    public static Date getApplicationDate() {
+        
+        if (UTIL_Namespace.isScratchOrg() && testingConfiguration != null) { 
+            return testingConfiguration.applicationDate;
+        }
+
+        try {
+            Date featureParameterDate = new UTIL_FeatureManagement().checkPackageDateValue(APPLICATION_DATE);
+            return featureParameterDate == null || DEFAULT_NON_APPLICATED_DATE == featureParameterDate ? null : featureParameterDate;
+        } catch (System.NoDataFoundException ex) {
+            return null;
+        }
+    }
+
+    /**
+     * @description Retrieves the Trial expiration date.
+     * On prod this is populated quering the Organization object.
+     * For QA purpose this values can be overriden if this is running on a Scratch org and there is static resource called GettingStartedTestingConfigurationJSON
+     * @return   DateTime with the expirationDate.
+     */
+    public static DateTime getExpirationDate() {
+        if (overrideValuesForTesting) {
+            return  testingConfiguration.trialExpiration;
+        } else {
+            return org.TrialExpirationDate;
+        }
+    }
+
+    /**
+     * @description Returns a Map<String,Object> retrieved from a Static Resource called 'GettingStartedTestingConfigurationJSON'
+     * @return   TestingConfig testing configuration after parsing the Static Resource Json
+     * Examples: 
+     * {"Sandbox":false} simulates an Active org.
```

Comment:
These comments need updating to match the format of the `TestingConfiguration` inner class properties.